pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -305,6 +305,10 @@ type PipelineLimits struct {
 	// +kubebuilder:default=80
 	// +optional
 	BufferUsageLimit *uint32 `json:"bufferUsageLimit,omitempty" protobuf:"varint,3,opt,name=bufferUsageLimit"`
+	// Read timeout seconds for all the vertices in the pipeline, can be overridden by the vertex's limit settings
+	// +kubebuilder:default= "1s"
+	// +optional
+	ReadTimeout string `json:"readTimeout,omitempty" protobuf:"bytes,4,opt,name=readTimeout"`
```

Comment:
Initially, I thought it won't be marshall/unmarshall.  did you use it anywhere?
But it will marshall number to duration like http://play.golang.org/p/CZH9ThvhxK

Reply:
Use `*metav1.Duration`, it will be good.